### DISCUSSION TITLE
ESP32-cam lens module support

### DISCUSSION THREAD

**Discussion by insomniux** [arendst/Tasmota#20409] [Q&A] [answered]
First of all my great thanks to the developers for the Tasmota platform. 
I've set up some esp32-cam's which work out of the box. Normally the esp32-cam comes with an OV2640 lens, but there are also the OV5640 lenses which supposed to have som more resolution. I've tested 2 of these lenses under ESPHome and they work but under Tasmota they seem not to be recognized. With the OV2640, in the log's, I get the line ' OV2640 initialized'. With the OV5640 this logline is not present. 
Is the OV5640 not supported on purpose? Or does it need additional drivers to be written?

Second question is about the OV2640 lens orientation. While making an enclosure for the ESP32-cam I damaged a lens module for which I ordered some replacement lenses. Now I found out that not all lenses have the same orientation (vertically or horizontally flipped) and some lenses have 90° rotation. I have not found an option in Tasmota to rotate the image. Did I miss this somewhere or is it not possible to rotate the image at the software side?

**Comment by sfromis** [comment]
It is not about the lens, you can also get the OV2640 camera sensor with another lens.

The Tasmota camera driver was built for the "original" AI-Thinker ESP32-CAM supporting OV2640 or OV7670. Work would be needed to support additional cameras.

To rotate the camera 90 degrees, use your hand. :wink:

**Reply by insomniux** [reply]
sure you're right, I used the wrong word, I should have written camera, as I'm actually interested in higher resolution rather than different lens types like wide angle or tele...

From your reply to my question about image rotation, should I conclude that it is not possible to do this, given the ESP32-cam board? It would be convenient to solve this at software level rather than to design a new enclosure...

**Reply by sfromis** [reply]
Sure, I'd have suggested it if there was another way available....

**Comment by Jason2866** [comment, marked as answer]
To save flash space, support for other cameras than OV2640 has been removed in Tasmota IDF/Arduino.
